dockstore-common/src/main/java/io/dockstore/common/yaml/DockstoreYamlHelper.java

```diff
@@ -194,6 +198,26 @@ private static Validator createValidator() {
         return validatorFactory.getValidator();
     }
 
+    /**
+     * Decide whether a gitReference is excluded, given a workflow/service's filters
+     * @param gitRefPath Path.of(gitReference) for glob matching with PathMatcher
+     * @param filters Filters specified for a workflow/service in .dockstore.yml
+     * @return
+     */
+    public static boolean filterGitReference(final Path gitRefPath, final List<String> filters) {
+        return filters.isEmpty() || filters.stream().anyMatch(filter -> {
+            String matcherString;
+            // Use regex if filter string is surrounded by /
+            if (filter.matches("^\\/.*\\/$")) {
+                matcherString = "regex:^refs\\/(heads\\/|tags\\/|)" + filter.substring(1, filter.length() - 1);
```

Comment:
I'm getting confused by this automatic start to the regex. It may be OK, and I just may not be understanding it, so speculating at this point:

As an example, how would I specify via RegEx all tags that start with `Release`?  As an end-user, I would be inclined to to write `/^tags\/Release.*/`, but I don't think the double caret in here would work in conjunction the existing caret in `^refs`.

Should I instead write `/^refs\/tags\/Release.*/`? Although I'd still have the same problem.

Reply:
> I'm getting confused by this automatic start to the regex. It may be OK, and I just may not be understanding it, so speculating at this point:

So the automatic starts for both the regex & glob are equivalent to eachother and are required to achieve the behaviour we're looking for.
For instance, with regex we want `/develop/` to match `refs/heads/develop`.
But we also want to specify something like `/tags\/.*/` (tags only) to match `refs/tags/anything` and not `refs/heads/anything`.

> As an example, how would I specify via RegEx all tags that start with `Release`?

`/tags\/Release.*/`

> As an end-user, I would be inclined to to write `/^tags\/Release.*/`, but I don't think the double caret in here would work in conjunction the existing caret in `^refs`.

The caret wouldn't work with the current prepended regex; `/^tags\/Release.*/` would become `^refs\\/(heads\\/|tags\\/|)^tags\/Release.*`.

I think that as long as we clearly document the precise behaviour & have some usage examples in our documentation, users can easily find out what to expect & achieve desired behaviour.